src/main/java/hudson/plugins/ec2/ssh/verifiers/HostKeyHelper.java

```diff
@@ -57,6 +64,24 @@ public static HostKeyHelper getInstance() {
         return INSTANCE;
     }
 
+    /**
+     * Converts a Java {@link PublicKey} to a {@link HostKey} for SSH verification.
+     * Uses the key's encoded form and BouncyCastle utilities to produce the SSH format.
+     *
+     * @param serverKey the public key to convert
+     * @return a {@link HostKey} representing the SSH-formatted key, or {@code null} if the algorithm is unsupported
+     * @throws IOException if the key cannot be processed
+     */
+    @CheckForNull
+    public HostKey getHostKey(@NonNull PublicKey serverKey) throws IOException {
+        String sshAlgorithm = KeyHelper.getSshAlgorithm(serverKey);
+        if (sshAlgorithm == null) {
+            return null;
+        }
+        AsymmetricKeyParameter parameters = PublicKeyFactory.createKey(serverKey.getEncoded());
+        return new HostKey(serverKey.getAlgorithm(), OpenSSHPublicKeyUtil.encodePublicKey(parameters));
```

Comment:
This should not be `serverKey.getAlgorithm()` but `sshAlgorithm`. This is causing an issue https://issues.jenkins.io/browse/JENKINS-76077